Add documentation for comparing json and ordered_json (#3599)
nlohmann/json: docs/mkdocs/docs/api/basic_json/operator_eq.md

    }
    ```

!!! note "Comparing different `basic_json` specializations"

>     Comparing different `basic_json` specializations can have surprising results. For instance, comparing the JSON
>     objects

Reviewer: ```suggestion
    Comparing different `basic_json` specializations can have surprising effects. For instance, the result of comparing
    the JSON objects
```

    ```json
    {
       "version": 1,
       "type": "integer"
    }
    ```

    and

    ```json
    {
       "type": "integer",
       "version": 1
    }
    ```

>     depends on whether [`nlohmann::json`](../json.md) order [`nlohmann::ordered_json`](../ordered_json.md) is used:

Reviewer: "or" oder "order"? :-)

```suggestion
    depends on whether [`nlohmann::json`](../json.md) or [`nlohmann::ordered_json`](../ordered_json.md) is used:
```

>     depends on whether [`nlohmann::json`](../json.md) order [`nlohmann::ordered_json`](../ordered_json.md) is used:

Reviewer: "or" oder "order"? :-)

```suggestion
    depends on whether [`nlohmann::json`](../json.md) or [`nlohmann::ordered_json`](../ordered_json.md) is used:
```
